filesystem/filesystem_windows.go

```diff
@@ -1,30 +1,133 @@
 package filesystem
 
 import (
-	utils "github.com/DataDog/gohai/windowsutils"
 	"strconv"
+	"syscall"
+	"unsafe"
 )
 
-func getFileSystemInfo() (interface{}, error) {
+type Handle uintptr
+
+const InvalidHandle Handle = Handle(^Handle(0))
+const ERROR_MORE_DATA syscall.Errno = 234
+
+// this would probably go in a common utilities rather than here
 
-	volumes, err := utils.WindowsWMIMultilineCommand("VOLUME", "Name", "Capacity", "DriveLetter")
-	if err != nil {
-		return nil, err
+func convert_windows_string_list(winput []uint16) []string {
+	var retstrings []string
+	var rsindex = 0
+
+	retstrings = append(retstrings, "")
+	for i := 0; i < (len(winput) - 1); i++ {
+		if winput[i] == 0 {
+			if winput[i+1] == 0 {
+				return retstrings
+			}
+			rsindex++
+			retstrings = append(retstrings, "")
+			continue
+		}
+		retstrings[rsindex] += string(rune(winput[i]))
 	}
-	var fileSystemInfo = make([]interface{}, len(volumes))
-	for i, volume := range volumes {
-		var capacity string
-		intCapacity, err := strconv.ParseInt(volume["Capacity"], 10, 64)
-		if err != nil {
-			capacity = "Unknown"
-		} else {
-			capacity = strconv.FormatInt(intCapacity/1024.0, 10)
+	return retstrings
+}
+
+// as would this
+func convert_windows_string(winput []uint16) string {
+	var retstring string
+	for i := 0; i < len(winput); i++ {
+		if winput[i] == 0 {
+			break
 		}
-		fileSystemInfo[i] = map[string]interface{}{
-			"name":       volume["Name"],
-			"kb_size":    capacity,
-			"mounted_on": volume["DriveLetter"],
+		retstring += string(rune(winput[i]))
+	}
+	return retstring
+}
+func getDiskSize(vol string) (size uint64, freespace uint64) {
+	var mod = syscall.NewLazyDLL("kernel32.dll")
+	var getDisk = mod.NewProc("GetDiskFreeSpaceExW")
+	var sz uint64
+	var fr uint64
+	status, _, _ := getDisk.Call(uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr(vol))),
+		uintptr(0),
+		uintptr(unsafe.Pointer(&sz)),
+		uintptr(unsafe.Pointer(&fr)))
+	if status == 0 {
+		return 0, 0
+	}
+	return sz, fr
+}
+func getMountPoints(vol string) []string {
+	var mod = syscall.NewLazyDLL("kernel32.dll")
+	var getPaths = mod.NewProc("GetVolumePathNamesForVolumeNameW")
+	var tmp uint32
+	var objlistsize uint32 = 0x0
+	var retval []string
+
+	status, _, errno := getPaths.Call(uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr(vol))),
+		uintptr(unsafe.Pointer(&tmp)),
+		2,
+		uintptr(unsafe.Pointer(&objlistsize)))
+
+	if status != 0 || errno != ERROR_MORE_DATA {
+		// unexpected
+		return retval
```

Comment:
do you think this error is worth returning as an error so that it gets logged somewhere or should we ignore it?